test/basic_test.rb

```diff
@@ -14,6 +14,13 @@ def test_timeout
     end
   end
 
+  def test_timeout_with_proc
+    self.settings = { service_timeout: proc { 5 } }
+    assert_raises(Rack::Timeout::RequestTimeoutError) do
+      get "/sleep"
+    end
+  end
```

Comment:
Would also be nice to have tests that do something different based on the env parameter to #call